### PR TITLE
[MM-45555] - Update pricing modal to reflect updated design

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
Update pricing modal design

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45555
https://mattermost.atlassian.net/browse/MM-45879

#### Screenshots
Check spinwick for different pricing modal states

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by neallred — CHANGES_REQUESTED** [review]
Looks good overall! Most of the things I found were style-related.

File: components/pricing_modal/self_hosted_content.tsx [hunk 1]
```diff
@@ -235,7 +214,11 @@ function SelfHostedContent(props: ContentProps) {
                         customButtonDetails={(!isPostSelfHostedEnterpriseTrial && isAdmin) ? (
                             trialButton()
                         ) : undefined}
-                        planDisclaimer={isPostSelfHostedEnterpriseTrial ? undefined : <StartTrialCaution/>}
```
Comment (marianunez) [inline components/pricing_modal/self_hosted_content.tsx:238]: We probably still need the legal verbage to start a trial on both cloud and self-hosted. @annelieseklein @johndavidlugtu we need to include this in the design as well

**Review by marianunez — CHANGES_REQUESTED** [review]
Changes overall LGTM, some changes need to be done to copies and legal text for trial:

**Comment by marianunez** [comment]
Two issues I noticed: 

1. I upgraded to Professional but actually was upgraded to Enterprise. 

2. Most likely an existing bug but once you upgrade to Enterprise (even though the intention was Professional), then go to System Console > Subscription page, click View Plans, the Pricing Modal allows you to click `Upgrade` again because it's enabled. 

Screen Shot 2022-07-27 at 6 55 31 PM

File: components/pricing_modal/self_hosted_content.tsx (hunk 1 above)
Comment (johndavidlugtu) [reply components/pricing_modal/self_hosted_content.tsx:238]: Heads up that the legal copy from this one can be used: https://www.figma.com/file/rWB5bbi3DvpmVwVMc2xfOH/Update-Plans?node-id=125%3A60793.

Note that we need to use the mattermost.com/pl/software-evaluation-agreement for this -- so we need to ping Marketing

File: components/pricing_modal/content.tsx [hunk 2]
```diff
@@ -213,75 +203,74 @@ function Content(props: ContentProps) {
...
+                        planExtraInformation={(!isAdmin && (isStarter || isEnterpriseTrial)) ? <NotifyAdminCTA/> : undefined}
+                        buttonDetails={{
+                            action: openPurchaseModal,
+                            text: formatMessage({id: 'pricing_modal.btn.upgrade', defaultMessage: 'Upgrade'}),
+                            disabled: !isAdmin || isProfessional,
+                            customClass: isPostTrial ? ButtonCustomiserClasses.special : ButtonCustomiserClasses.active,
+                        }}
                         briefing={{
-                            title: formatMessage({id: 'pricing_modal.briefing.title', defaultMessage: 'Top features'}),
+                            title: formatMessage({id: 'pricing_modal.briefing.professional.title', defaultMessage: 'No limits on your team’s usage'}),
```
[the remark this answers is not in the document]
Comment (johndavidlugtu) [reply components/pricing_modal/content.tsx:235]: Yeah it should be `Top Features` for each plan actually. Refer to this [one](https://docs.google.com/document/d/1i9iuyyV-PyuhIw19gxnCEbFfe6IE2fnWOoMThiz926o/edit#heading=h.g4bt3wv5ap6y) for the latest copy.

File: components/pricing_modal/content.tsx [hunk 3]
```diff
@@ -170,24 +163,21 @@ function Content(props: ContentProps) {
...
-                    <div className='self-hosted-svg-right'>
-                        <ManSvg/>
-                    </div>
                     <Card
                         id='starter'
-                        topColor='#9DA7B8'
+                        topColor='#339970'
                         plan='Starter'
+                        planSummary={formatMessage({id: 'pricing_modal.planSummary.starter', defaultMessage: 'Increased productivity for small teams'})}
                         price={formatMessage({id: 'pricing_modal.price.free', defaultMessage: 'Free'})}
```
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply components/pricing_modal/content.tsx:171]: I think the design recently changed. Updated.

File: components/common/plan_label/index.tsx [hunk 4]
```diff
@@ -22,14 +32,38 @@ color: ${(props) => props.color};
...
 function PlanLabel(props: PlanLabelProps) {
+    const {formatMessage} = useIntl();
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const subscription = useSelector(selectCloudSubscription);
+    const license = useSelector(getLicense);
+    const isSelfHostedEnterpriseTrial = license.IsTrial === 'true';
+
+    let text = props.text;
+
+    if (props.renderLastDaysOnTrial && isCloud && subscription?.is_free_trial === 'true') {
```
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply components/common/plan_label/index.tsx:43]: PlanLabel is also used on the purchase_modal where want to always show the label with text 'MOST POPULAR' regardless of if they're on a trial

File: components/pricing_modal/content.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply components/pricing_modal/content.tsx:235]: @johndavidlugtu should we do the same for the self-hosted modal?

File: components/pricing_modal/self_hosted_content.tsx (hunk 1 above)
Comment (AGMETEOR) [reply components/pricing_modal/self_hosted_content.tsx:238]: Added it back 
Screenshot 2022-07-28 at 13 46 28

**Comment by AGMETEOR** [comment]
> Cws@10102020

Hey Maria. I have failed to reproduce issue 1. On the current spinwick, I upgraded to professional successfully. I have fixed issue 2

File: components/pricing_modal/self_hosted_content.tsx (hunk 1 above)
Comment (AGMETEOR) [reply components/pricing_modal/self_hosted_content.tsx:238]: > Heads up that the legal copy from this one can be used: https://www.figma.com/file/rWB5bbi3DvpmVwVMc2xfOH/Update-Plans?node-id=125%3A60793.
> 
> Note that we need to use the mattermost.com/pl/software-evaluation-agreement for this -- so we need to ping Marketing

@johndavidlugtu this link `https://mattermost.com/pl/software-evaluation-agreement` is dead

Comment (johndavidlugtu) [reply components/pricing_modal/self_hosted_content.tsx:238]: @annelieseklein thoughts on the design (see screenshot)?

@AGMETEOR feel free to ping in the Marketing channel or to Emily Dela Cruz directly

File: components/pricing_modal/content.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (johndavidlugtu) [reply components/pricing_modal/content.tsx]: @AGMETEOR not sure if I'm missing context, but `Top features` is the sub-header for each plan for both Cloud and Self-hosted

File: components/pricing_modal/self_hosted_content.tsx (hunk 1 above)
Comment (marianunez) [reply components/pricing_modal/self_hosted_content.tsx:238]: @emdecr heads up on link above

File: components/pricing_modal/content.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply components/pricing_modal/content.tsx]: Alright

File: components/pricing_modal/self_hosted_content.tsx (hunk 1 above)
Comment (emdecr) [reply components/pricing_modal/self_hosted_content.tsx:238]: @johndavidlugtu @marianunez  Can def set up that redirect up - just to clarify: we want `/pl/software-evaluation-agreement` to go to https://mattermost.com/software-evaluation-agreement/ ?

Comment (AGMETEOR) [reply components/pricing_modal/self_hosted_content.tsx:238]: Yes that's correct

**Review by neallred — APPROVED** [review]
LGTM, thanks @AGMETEOR !

**Comment by marianunez** [comment]
Noticed the `Start Trial` is showing even though they are in Professional - starting a trial gives an error in this case (we may be preventing it in the backend) 

@johndavidlugtu is it expected for them to start a trial when already in Professional ? 
Screen Shot 2022-07-28 at 3 53 43 PM

**Comment by marianunez** [comment]
In self-managed we should surface the same `View Plans` button we have in the subscription card but in the License page (see below) - I've added a ticket to address that separately from this PR https://mattermost.atlassian.net/browse/MM-46060

Screen Shot 2022-07-28 at 4 09 40 PM

**Review by marianunez — APPROVED** [review]
Overall LGTM, let a comment regarding the Start Trial during Professional. Also, could we update the E2E tests to include the new cases introduced here?

**Comment by johndavidlugtu** [comment]
> Noticed the `Start Trial` is showing even though they are in Professional - starting a trial gives an error in this case (we may be preventing it in the backend)
> 
> @johndavidlugtu is it expected for them to start a trial when already in Professional ? Screen Shot 2022-07-28 at 3 53 43 PM
> 
> Also, @annelieseklein thoughts on the way the legal text is pushing down the `Top Features` section compared to the others?

If they haven't activated a Trial before, they should be able to start an Enterprise trial while on Starter or Professional.

**Comment by marianunez** [comment]
> If they haven't activated a Trial before, they should be able to start an Enterprise trial while on Starter or Professional.

Now I remember the discussions around this, activating a trial during Professional is not as straight forward because 1- we currently don't have a way at the end of the trial to downgrade back to Professional, we currently automatically downgrade to Starter. 2- There would be some billing implication that we would need to sort out. 
@johndavidlugtu if we still want to support this, we need to hash out those details and would be covered in a separate work. Feel free to open a ticket if you would still consider this a valuable addition. 

In the meantime for the purposes of this PR, @AGMETEOR we need to disable the `Start trial` if they are currently in Professional as it is not yet supported.

**Comment by johndavidlugtu** [comment]
> > If they haven't activated a Trial before, they should be able to start an Enterprise trial while on Starter or Professional.
> 
> Now I remember the discussions around this, activating a trial during Professional is not as straight forward because 1- we currently don't have a way at the end of the trial to downgrade back to Professional, we currently automatically downgrade to Starter. 2- There would be some billing implication that we would need to sort out. @johndavidlugtu if we still want to support this, we need to hash out those details and would be covered in a separate work. Feel free to open a ticket if you would still consider this a valuable addition.
> 
> In the meantime for the purposes of this PR, @AGMETEOR we need to disable the `Start trial` if they are currently in Professional as it is not yet supported.

Thanks for the context @marianunez ! It's definitely something we should support at some point. That said, I don't think this is a priority right now

**Review by stevemudie — APPROVED** [review]
Looks good @AGMETEOR!

**Comment by annelieseklein** [comment]
Screen Shot 2022-08-02 at 8 10 17 AM
@AGMETEOR The legalese should go actually above the Top features in this new plans modal. Also can we please reduce the spacing between the legalese text?

**Comment by annelieseklein** [comment]
@AGMETEOR also the grey part of the legalese text should be 56% Center channel text - so probably just a little lighter than what you have it here

**Review by annelieseklein — APPROVED** [review]
Great work!

**Pull request merged**